src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.Import.cs

```diff
@@ -35,9 +47,70 @@ public static CngKey Import(byte[] keyBlob, CngKeyBlobFormat format, CngProvider
 
             SafeNCryptProviderHandle providerHandle = provider.OpenStorageProvider();
             SafeNCryptKeyHandle keyHandle;
-            ErrorCode errorCode = Interop.NCrypt.NCryptImportKey(providerHandle, IntPtr.Zero, format.Format, IntPtr.Zero, out keyHandle, keyBlob, keyBlob.Length, 0);
-            if (errorCode != ErrorCode.ERROR_SUCCESS)
-                throw errorCode.ToCryptographicException();
+            ErrorCode errorCode;
+            
+            if (curve == null)
+            {
+                errorCode = Interop.NCrypt.NCryptImportKey(providerHandle, IntPtr.Zero, format.Format, IntPtr.Zero, out keyHandle, keyBlob, keyBlob.Length, 0);
+                if (errorCode != ErrorCode.ERROR_SUCCESS)
+                {
+                    Exception e = errorCode.ToCryptographicException();
+                    if (errorCode == ErrorCode.NTE_NOT_SUPPORTED &&
+                        (format == CngKeyBlobFormat.EccFullPrivateBlob || format == CngKeyBlobFormat.EccFullPrivateBlob))
+                    {
+                        // Attempt to import on < Windows 10
+                        throw new PlatformNotSupportedException(SR.Cryptography_InvalidCurveOid, e);
+                    }
+                    throw e;
+                }
+            }
+            else
+            {
+                // Call with Oid.FriendlyName because .Value will result in an invalid parameter error
+                Debug.Assert(curve.Value.IsNamed);
+                string curveName = curve.Value.Oid.FriendlyName; 
+                using (SafeUnicodeStringHandle safeCurveName = new SafeUnicodeStringHandle(curveName))
+                {
+                    var desc = new Interop.BCrypt.BCryptBufferDesc();
+                    var buff = new Interop.BCrypt.BCryptBuffer();
+
+                    IntPtr descPtr = IntPtr.Zero;
+                    IntPtr buffPtr = IntPtr.Zero;
+                    try
+                    {
+                        descPtr = Marshal.AllocHGlobal(Marshal.SizeOf(desc));
+                        buffPtr = Marshal.AllocHGlobal(Marshal.SizeOf(buff));
+                        buff.cbBuffer = (curveName.Length + 1) * 2;
```

Comment:
I'd, personally, leave a comment like // Add one for the null terminator

Random `+ 1`s tend to catch my skeptical eye.